components/patient-data/migrations/src/main/java/org/phenotips/data/internal/R71490PhenoTips1280DataMigration.java

```diff
@@ -226,10 +262,26 @@ private void migrateGenes(XWikiDocument doc, DocumentReference oldGenesClassRefe
             } else if (geneComments != null) {
                 String commentAppend = "Automatic migration: \ngene was duplicated in the " + status + " gene section.";
                 commentAppend += "\nOriginal comment: \n" + geneComments;
-                updateComment(geneName, doc, commentAppend, geneClassReference);
+                updateComment(geneName, doc, commentAppend, this.geneClassReference);
             }
         }
         doc.removeXObjects(oldGenesClassReference);
+
+        // fix for PT-3033: users occasionally used the "rejected genes" section in 1.2 only for comments
+        if (StringUtils.isNotBlank(freeComments.toString())) {
+            this.commentsFromEmptyGenes +=
+                (CANDIDATE_NAME.equals(status)) ? CANDIDATE_COMMENT_START : REJECTED_COMMENT_START
```

Comment:
This line means:

    if the status is candidate
    then append CANDIDATE_COMMENT_START
    else append REJECTED_COMMENT_START + freeComments

So `freeComments` are lost for candidate genes.